### PR TITLE
Avoid using internal API from package `org.graalvm.nativeimage.impl`

### PR DESCRIPTION
**Pull request by zakkak**
As discussed in https://github.com/quarkusio/quarkus/pull/27728#discussion_r971970749

### CODE REVIEW

**Comment by zakkak** [comment]
The removal of the remaining usages requires upstream changes, see https://github.com/oracle/graal/issues/5013.

**Comment by zakkak** [comment]
@jerboaa can you please review this?

**Review by galderz — APPROVED** [review]
LGTM but @jerboaa should have a second look since he was involved in the original discussions.

File: extensions/awt/runtime/src/main/java/io/quarkus/awt/runtime/graal/DarwinAwtFeature.java [hunk 1]
```diff
@@ -13,10 +12,8 @@ public class DarwinAwtFeature implements Feature {
     @Override
     public void afterRegistration(AfterRegistrationAccess access) {
         if (JavaVersionUtil.isJava17OrHigher()) {
-            final RuntimeClassInitializationSupport runtimeInit = ImageSingletons
-                    .lookup(RuntimeClassInitializationSupport.class);
-            final String reason = "Quarkus run time init for AWT in Darwin";
-            runtimeInit.initializeAtRunTime("sun.lwawt.macosx", reason);
+            // Quarkus run time init for AWT in Darwin
+            RuntimeClassInitialization.initializeAtBuildTime("sun.lwawt.macosx");
```
Comment (jerboaa) [inline extensions/awt/runtime/src/main/java/io/quarkus/awt/runtime/graal/DarwinAwtFeature.java:16]: Typo? `s/initializeAtBuildTime/initializeAtRunTime/`

**Review by jerboaa — CHANGES_REQUESTED** [review]
Looks mostly good. I think one unintentional change from runtime->buildtime sneaked in, though.

File: extensions/awt/runtime/src/main/java/io/quarkus/awt/runtime/graal/DarwinAwtFeature.java (hunk 1 above)
Comment (zakkak) [reply extensions/awt/runtime/src/main/java/io/quarkus/awt/runtime/graal/DarwinAwtFeature.java]: Good catch. Thanks @jerboaa

**Review by jerboaa — APPROVED** [review]
LGTM

**Pull request merged**